packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java

```diff
@@ -159,6 +162,30 @@ private static String getPathFromRemoteUri(final Context context, final Uri uri)
     return success ? file.getPath() : null;
   }
 
+  private static String getImageExtension(Context context, Uri uriImage) {
+    String extension = "";
+    Cursor cursor = null;
+
+    try {
+      cursor = context.getContentResolver().query(uriImage,
+              new String[]{MediaStore.MediaColumns.MIME_TYPE},
+              null, null, null);
+
+      if (cursor != null && cursor.moveToNext()) {
+        String mimeType = cursor.getString(0);
+
+        extension = MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType);
+      }
+    } catch (Exception ignored) {
+
```

Comment:
ignoring exception seems odd. Do we not have to resolve this if there is an exception?

Reply:
if exception was thrown, final file name will be without extension.